### PR TITLE
Consistent rpc response case

### LINKED ISSUE #680: Consistent RPC response cases
We have a bunch of different casings for rpc responses.

Let's fix them before people build stuff that rely on the current impl.

UPD(Esad):

> it’s mostly in the responses. citrea_syncStatus, ledger_getSoftConfirmationByNubmer, eth_blockNumber ...

### PR DESCRIPTION
**Pull request by yaziciahmet**
# Description

Made all RPC responses in camel case, even the possibly unused ones.

## Linked Issues
- Fixes #680

### CODE REVIEW

File: crates/sovereign-sdk/rollup-interface/src/node/rpc/mod.rs [hunk 1]
```diff
@@ -695,6 +698,80 @@ pub mod utils {
             deserializer.deserialize_str(HexStrVisitor(PhantomData))
         }
     }
+
+    /// Serialization and deserialization logic for `0x`-prefixed hex strings that are wrapped with Option.
```
Comment (kpp) [inline crates/sovereign-sdk/rollup-interface/src/node/rpc/mod.rs:702]: Use TxHex that already exists in the project if you can.

Comment (yaziciahmet) [reply crates/sovereign-sdk/rollup-interface/src/node/rpc/mod.rs]: Done https://github.com/chainwayxyz/citrea/pull/867/commits/4c78cd7ba2a84b67a8cd27d98d3a45c4b8fbd7f6

**Review by eyusufatik — APPROVED** [review]

**Review by kpp — APPROVED** [review]
LGTM.

@eyusufatik maybe we can fix https://github.com/chainwayxyz/citrea/issues/716 here?

**Comment by eyusufatik** [comment]
> https://github.com/chainwayxyz/citrea/issues/838

let's merge this 

that can be done in any other pr haha

**Pull request merged**